File: Core Java/accountPolymorphism/src/com/techlabs/account/Account.java
package com.techlabs.account;

public abstract class Account {

	private int accountNumber;
	private String name;
	protected double balance;

	public Account(int account, String name, double balance) {
		this.accountNumber = account;
		this.name = name;
		this.balance = balance;
	}

	public void deposit(double balance) {
		this.balance = this.balance + balance;
	}

	public abstract void withdraw(double balance);

	public int getAccount() {
		return accountNumber;
	}

	public String getName() {
		return name;
	}

	public double getBalance() {
		return balance;
	}
}

File: DOTNET/C#/AccountPolymorphism/AccountPolymorphism/Account.cs
using System;

namespace AccountPolymorphism
{
    public abstract class Account
    {
        private readonly int _accountNumber;
        private readonly String _name;
        protected double balance;

        public Account(int accountNumber, String name, double balance)
        {
            _accountNumber = accountNumber;
            _name = name;
            this.balance = balance;
        }

        public void Deposit(double balance)
        {
            this.balance = this.balance + balance;
        }

        public abstract void WithDraw(double balance);

        public int AccountNumber
        {
            get
            {
                return _accountNumber;
            }
        }

        public String Name
        {
            get
            {
                return _name;
            }
        }

        public double Balance
        {
            get
            {
                return balance;
            }
        }
    }
}
